### PR TITLE
[DROOLS-6064] Inconsistent behavior when accumulate function returns …

### PR DESCRIPTION
**Pull request by tkobayas**
…null

- WIP. This should be disabled by default

https://issues.redhat.com/browse/DROOLS-6064

**JIRA**: 

https://issues.redhat.com/browse/DROOLS-6064

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-core/src/main/java/org/drools/core/phreak/PhreakAccumulateNode.java [hunk 1]
```diff
@@ -48,6 +48,9 @@
 * To change this template use File | Settings | File Templates.
 */
 public class PhreakAccumulateNode {
+
+    public static boolean ALLOW_NULL_PROPAGATION = true; // default will be false for backward compatibility
```
Comment (tkobayas) [inline drools-core/src/main/java/org/drools/core/phreak/PhreakAccumulateNode.java:52]: This is still hard-coded. Will make it configurable (default will be false)

Comment (mariofusco) [reply drools-core/src/main/java/org/drools/core/phreak/PhreakAccumulateNode.java:52]: @tkobayas sorry, not understanding this comment. Default will be `false` but you temporarily set it to `true`?

Comment (tkobayas) [reply drools-core/src/main/java/org/drools/core/phreak/PhreakAccumulateNode.java:52]: @mariofusco I temporarily set this true in order to test this change. Now I will work on the configuration. Exactly I'll do what you said in the email:
> That said I believe that we will need a configuration option for this, something that could be defaulted to false for backward compatibility now, but we could possibly turn it to true for drools 8.

**Comment by tkobayas** [comment]
Jenkins run fdb

**Comment by tkobayas** [comment]
FDB error
```
org.optaplanner.core.api.score.stream.AdvancedGroupByConstraintStreamTest.collectedFilteredRecollected[2]

Failing for the past 2 builds (Since Unstable#125 )
Took 7 ms.
Error Message
Exception executing consequence for rule "testConstraintName" in org.optaplanner.core.impl.testdata.domain.score.lavish: java.lang.NullPointerException
Stacktrace
Exception executing consequence for rule "testConstraintName" in org.optaplanner.core.impl.testdata.domain.score.lavish: java.lang.NullPointerException
	at org.drools.core.runtime.rule.impl.DefaultConsequenceExceptionHandler.handleException(DefaultConsequenceExceptionHandler.java:39)
	at org.drools.core.common.DefaultAgenda.handleException(DefaultAgenda.java:1306)
	at org.drools.core.phreak.RuleExecutor.innerFireActivation(RuleExecutor.java:455)
	at org.drools.core.phreak.RuleExecutor.fireActivation(RuleExecutor.java:395)
	at org.drools.core.phreak.RuleExecutor.fire(RuleExecutor.java:151)
	at org.drools.core.phreak.RuleExecutor.evaluateNetworkAndFire(RuleExecutor.java:103)
	at org.drools.core.concurrent.AbstractRuleEvaluator.internalEvaluateAndFire(AbstractRuleEvaluator.java:33)
	at org.drools.core.concurrent.SequentialRuleEvaluator.evaluateAndFire(SequentialRuleEvaluator.java:43)
	at org.drools.core.common.DefaultAgenda.fireLoop(DefaultAgenda.java:1116)
	at org.drools.core.common.DefaultAgenda.internalFireAllRules(DefaultAgenda.java:1063)
	at org.drools.core.common.DefaultAgenda.fireAllRules(DefaultAgenda.java:1055)
	at org.drools.core.impl.StatefulKnowledgeSessionImpl.internalFireAllRules(StatefulKnowledgeSessionImpl.java:1346)
	at org.drools.core.impl.StatefulKnowledgeSessionImpl.fireAllRules(StatefulKnowledgeSessionImpl.java:1337)
	at org.drools.core.impl.StatefulKnowledgeSessionImpl.fireAllRules(StatefulKnowledgeSessionImpl.java:1321)
	at org.optaplanner.core.impl.score.stream.drools.DroolsConstraintSession.calculateScore(DroolsConstraintSession.java:62)
	at org.optaplanner.core.impl.score.director.stream.ConstraintStreamScoreDirector.calculateScore(ConstraintStreamScoreDirector.java:74)
	at org.optaplanner.core.api.score.stream.AbstractConstraintStreamTest.assertScore(AbstractConstraintStreamTest.java:84)
	at org.optaplanner.core.api.score.stream.AdvancedGroupByConstraintStreamTest.collectedFilteredRecollected(AdvancedGroupByConstraintStreamTest.java:127)
	...
Caused by: java.lang.NullPointerException
	at org.drools.core.reteoo.FromNodeLeftTuple.getAccumulatedObjects(FromNodeLeftTuple.java:100)
	at org.drools.core.common.AgendaItem.getObjectsDeep(AgendaItem.java:79)
	at org.drools.core.reteoo.RuleTerminalNodeLeftTuple.getObjectsDeep(RuleTerminalNodeLeftTuple.java:359)
	at org.optaplanner.core.api.score.holder.AbstractScoreHolder.extractJustificationList(AbstractScoreHolder.java:209)
	at org.optaplanner.core.api.score.holder.AbstractScoreHolder.registerConstraintMatch(AbstractScoreHolder.java:132)
	at org.optaplanner.core.api.score.buildin.simple.SimpleScoreHolder.addConstraintMatch(SimpleScoreHolder.java:133)
	at org.optaplanner.core.api.score.buildin.simple.SimpleScoreHolder.lambda$configureConstraintWeight$1(SimpleScoreHolder.java:58)
	at org.optaplanner.core.api.score.buildin.simple.SimpleScoreHolder.impactScore(SimpleScoreHolder.java:120)
	at org.optaplanner.core.impl.score.stream.drools.common.DroolsCondition.impactScore(DroolsCondition.java:140)
	at org.optaplanner.core.impl.score.stream.drools.uni.DroolsUniCondition.lambda$completeWithScoring$18bbbaa0$1(DroolsUniCondition.java:302)
	at org.drools.model.functions.Block3$Impl.execute(Block3.java:40)
	at org.drools.modelcompiler.consequence.LambdaConsequence.evaluate(LambdaConsequence.java:76)
	at org.drools.core.phreak.RuleExecutor.innerFireActivation(RuleExecutor.java:448)
	... 124 more
```

**Comment by triceo** [comment]
@tkobayas This is likely unrelated to your commits. Mario is already investigating this failure, was introduced on Thursday.

**Comment by tkobayas** [comment]
> @tkobayas This is likely unrelated to your commits. Mario is already investigating this failure, was introduced on Thursday.

@triceo Thanks for letting me know :)

**Comment by tkobayas** [comment]
superseded by https://github.com/kiegroup/drools/pull/3481

**Pull request closed without merging**